### ISSUE TITLE
Question about inference

### ISSUE THREAD

**Issue by REXWindW** [Gift-OYS/MMSite#1]
Thank you for sharing your excellent work and the implementation! I have a question related to the inference process described in your paper:

In the paper, you mentioned that the input text during inference is generated by the Prot2Text model. Could you explain how this text is segmented into different attributes to fit the MACross? 

I've noticed that in model.py:
```
        if not test:
            anchor_text_outputs = []
            for i in range(len(anchor_text_input_ids)):
                input_id = anchor_text_input_ids[i]
                attention_mask = anchor_text_attention_mask[i]
                text_output = self._get_model_output(input_id, attention_mask, 'text')
                anchor_text_outputs.append(text_output)
        else:
            anchor_text_branch_output_o = self._get_model_output(anchor_text_input_ids, anchor_text_attention_mask, 'text')

        anchor_seq_outputs = self._get_model_output(anchor_seq_input_ids, anchor_seq_attention_mask, 'seq')
        anchor_seq_outputs = anchor_seq_outputs[0]

        if not test:
            anchor_text_branch_output_o = self._get_text_branch_output(anchor_text_outputs)
            anchor_text_branch_output = anchor_text_branch_output_o
        else:
            anchor_text_branch_output = anchor_text_branch_output_o[0]

        anchor_seq_branch_output = self.project(anchor_seq_outputs)
```
It seems that during train, the `anchor_text_input_ids` is a list, while in test it is only one ids(encoded descriptions generated by Prot2Text)
```
def _get_text_branch_output(self, text_outputs):
        texts_output = [self.texts_encoder[i](text_outputs[i][0]) for i in range(self.num_attr)]
        texts_output_cls = [texts_output[idx][:, 0, :].unsqueeze(1) for idx in range(len(texts_output)) if idx != 3]
        texts_output_cls = torch.cat(texts_output_cls, dim=1)
        texts_output_cls = self.text_suffix_transformer(texts_output_cls)
        text_func = texts_output[3]
        x = texts_output_cls
        for i in range(4):
            _x = x
            x = self.text_crosses[i](x, text_func, text_func)
            x = self.norms[i](x[0] + _x)
        text_branch_output = x
        return text_branch_output
```
It seems that during testing, the MACross module is skipped, and the outputs of PubMedBERT-abs is used to replace the MACross's Output. Would this cause a mismatch between the conditions during training and testing?

Thank you very much for your time and effort, and I appreciate your guidance!

**Comment by Gift-OYS** [comment]
Thank you for your insightful question!
﻿
You're correct. During training, textual descriptions are segmented and processed via the MACross module for fine-grained alignment with the protein sequence. However, during inference, Prot2Text generates a single textual description, which is directly processed by PubMedBERT, skipping the MACross module.
﻿
This discrepancy is mitigated by the soft-label alignment during training, ensuring that sequence and textual representations align closely in the latent space. Additionally, the STEnc module refines alignment during both phases, reducing the impact of skipping MACross in inference.
﻿
While this approach works well in practice, future improvements could involve refining Prot2Text to generate attribute-specific outputs or introducing a post-processing step to better align with MACross requirements.
﻿
Hope this clarifies your concern.
